src/observer.ts

```diff
@@ -1,15 +1,40 @@
 import { EventEmitter } from 'eventemitter3';
+import HlsEvents from './events';
+
+type Arguments < T > = [T] extends [ (...args: infer U) => any ]
+  ? U
+  : [T] extends [void] ? [] : [T];
+
+interface TypedEventEmitter<Events> {
+  addListener<E extends keyof Events> (event: E, listener: Events[E]): this
+  on<E extends keyof Events> (event: E, listener: Events[E]): this
+  once<E extends keyof Events> (event: E, listener: Events[E]): this
+
+  removeAllListeners<E extends keyof Events> (event: E): this
+  removeListener<E extends keyof Events> (event: E, listener: Events[E]): this
+
+  emit<E extends keyof Events> (event: E, ...args: Arguments<Events[E]>): boolean
+  listeners<E extends keyof Events> (event: E): Function[]
```

Comment:
couldn't this be
```suggestion
  listeners<E extends keyof Events> (event: E): Array<Events[E]>
```
?

Reply:
Yes, I think that's correct. It would be an array of the functions from the interface.